### PR TITLE
Find forms hooks update

### PR DESCRIPTION
**Pull request by zacharymorel**
## Description
During down time, I refactored some of the class components to use hooks.

## Testing done
Ran unit and e2e tests, ran local to test functionality.

## Screenshots
Screen Shot 2021-05-19 at 3 03 48 PM

## Acceptance criteria
- [x] Test pass, app works as it does on production

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/find-forms/containers/SearchForm.jsx [hunk 1]
```diff
@@ -10,91 +10,81 @@ import {
...
+export const SearchForm = ({
+  fetchForms,
+  fetching,
+  useLighthouseSearchAlgo,
+}) => {
+  // Derive the current query params.
+  const queryParams = new URLSearchParams(window.location.search);
+  // Derive the query.
+  const query = queryParams.get('q') || '';
+  const [queryState, setQueryState] = useState({
```
Comment (kelsonic) [inline src/applications/find-forms/containers/SearchForm.jsx:22]: Any reason to keep queryState as an object and not just have query as a string? 🙂

File: src/applications/find-forms/containers/SearchResults.jsx [hunk 2]
```diff
@@ -23,62 +23,78 @@ import { FAF_SORT_OPTIONS, FAF_TEST_OPTION_CLOSEST_MATCH } from '../constants';
...
+  query,
+  results,
+  sortByPropertyName,
+  hasOnlyRetiredForms,
+  startIndex,
+  useLighthouseSearchAlgo,
+  updatePagination,
+  updateSortByPropertyName,
+}) => {
+  const prevProps = usePreviousProps({
```
Comment (kelsonic) [inline src/applications/find-forms/containers/SearchResults.jsx:60]: So the object we're passing here is a current ref? I'm a little confused by this but it's nice that we're moving to hooks!

File: src/applications/find-forms/containers/SearchResults.jsx [hunk 3]
```diff
@@ -23,62 +23,78 @@ import { FAF_SORT_OPTIONS, FAF_TEST_OPTION_CLOSEST_MATCH } from '../constants';
...
     // Ensure the start index is not greater than the total amount of results.
-    if (startIndex >= results.length) {
-      startIndex = results.length - 1;
+    if (startI >= results.length) {
+      startI = results.length - 1;
     }
 
     // Update the page and the new start index.
-    updatePagination(page, startIndex);
+    updatePagination(p, startI);
```
Comment (kelsonic) [inline src/applications/find-forms/containers/SearchResults.jsx:90]: Could we not abbreviate here and just spell out page and startIndex for clarity? 🙂

**Review by kelsonic — APPROVED** [review]

File: src/applications/find-forms/containers/SearchForm.jsx (hunk 1 above)
Comment (zacharymorel) [reply src/applications/find-forms/containers/SearchForm.jsx:22]: Oh that's a thing. I forgot you can do that. 
Old habits with state being an object.

File: src/applications/find-forms/containers/SearchResults.jsx (hunk 2 above)
Comment (zacharymorel) [reply src/applications/find-forms/containers/SearchResults.jsx:60]: Yeah, it's really confusing but it's the only way I found where we can Mirror the behavior. 
@kelsonic here's what I found in the React docs. 
https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state

**Pull request merged**